components/raftstore/src/store/fsm/peer.rs

```diff
@@ -2125,9 +2127,28 @@ where
         if meta.region_ranges.remove(&last_key).is_none() {
             panic!("{} original region should exists", self.fsm.peer.tag);
         }
-        // It's not correct anymore, so set it to None to let split checker update it.
-        self.fsm.peer.approximate_size = None;
         let last_region_id = regions.last().unwrap().get_id();
+
+        let new_region_count = regions.len() as u64;
+        // Roughly estimate the size and keys and then let split checker to update it.
+        self.fsm.peer.approximate_size =
+            self.fsm.peer.approximate_size.map(|x| x / new_region_count);
+        self.fsm.peer.approximate_keys =
+            self.fsm.peer.approximate_keys.map(|x| x / new_region_count);
+        if let Err(e) = self.ctx.split_check_scheduler.schedule(
+            SplitCheckTask::GetRegionApproximateSizeAndKeys {
+                region: self.fsm.peer.region().clone(),
+                pending_tasks: Arc::new(AtomicU64::new(1)),
+                cb: Box::new(move |_, _| {}),
+            },
+        ) {
+            error!(
+                "failed to schedule split check task";
+                "region_id" => self.fsm.region_id(),
+                "peer_id" => self.fsm.peer_id(),
+                "err" => ?e,
+            );
+        }
```

Comment:
If it's not leader, this task is very wasteful and the `approximate_size` & `approximate_key` are no need to be updated.

Reply:
And I think `self.fsm.peer.heartbeat_pd` should be called here if it's a leader.

Reply:
> If it's not leader, this task is very wasteful and the `approximate_size` & `approximate_key` are no need to be updated.

Oh, I didn't notice here in the last update. I will add some check later.

> And I think self.fsm.peer.heartbeat_pd should be called here if it's a leader.

`heartbeat_pd` is called a few lines before here.

Reply:
@gengliqi I have updated. PTAL.

Reply:
I think we can move these codes to the upper `if is_leader`.

Reply:
I have updated again. Please have a look.